app/models/metadatum.rb

```diff
@@ -101,6 +102,21 @@ def check_and_set_date_type
     errors.add(:raw_value, MetadataValidationErrors::INVALID_DATE)
   end
 
+  def check_and_set_location_type
+    # Based on our metadata structure, the location details selected by the user will end up in
+    # raw_value.
+    loc = JSON.parse(raw_value, symbolize_names: true)
+
+    # Set to existing Location or create a new one based on the external IDs. For the sake of not
+    # trusting user input, we'll potentially re-fetch location details based on the API and OSM IDs.
+    result = Location.find_or_create_by_api_ids(loc[:locationiq_id], loc[:osm_id], loc[:osm_type])
+
+    # Re-use number_validated_value column to be the foreign key to the Location table
+    self.number_validated_value = result.id
```

Comment:
Could you create a new column `location_id` instead of re-using an old column?

Even though it creates an extra column, it keeps things clean. Otherwise, in the future, every time we do anything with number_validated_value, we have to remember the special case where location also re-uses that column.

Reply:
sure will change